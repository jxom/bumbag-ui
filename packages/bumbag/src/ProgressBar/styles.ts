import { css, cssClass } from '../styled';
import { borderRadius, tint, palette, theme } from '../utils';

export const ProgressBar = (styleProps) => cssClass`
  border-radius: ${borderRadius('default')(styleProps)};
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: ${palette(`${styleProps.color}Tint`, { dark: `${styleProps.color}Shade` })(styleProps)};

  ${getSizeAttributes(styleProps)};

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

export const ProgressBarIndicator = (styleProps) => cssClass`
  height: 100%;
  background-color: ${palette(styleProps.color)(styleProps)};
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  width: ${styleProps.value || '0'}%;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;

function getSizeAttributes(styleProps) {
  const sizeAttributes = {
    small: css`
      height: 0.6rem;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.small`)(styleProps)};
      }
    `,
    default: css`
      & {
        ${theme(styleProps.themeKey, `styles.sizes.default`)(styleProps)};
      }
    `,
    medium: css`
      height: 1.5rem;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.medium`)(styleProps)};
      }
    `,
    large: css`
      height: 2rem;

      & {
        ${theme(styleProps.themeKey, `styles.sizes.large`)(styleProps)};
      }
    `,
  };
  return sizeAttributes[styleProps.size || 'default'];
}
