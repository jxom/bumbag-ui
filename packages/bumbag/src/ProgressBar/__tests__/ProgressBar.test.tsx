import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { ProgressBar } from '../ProgressBar';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<ProgressBar value={50} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<ProgressBar color="primary" value={50} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with maxValue', () => {
    const { container } = render(<ProgressBar maxValue={150} value={50} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with size', () => {
    const { container } = render(<ProgressBar size="large" value={50} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('hook', () => {
    it('should return with ProgressBar props', () => {
      const { result } = renderHook(() => ProgressBar.useProps({ value: 50 }));
      expect(result.current).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('ProgressBar.base should render correctly', () => {
    const { container } = render(
      <ProgressBar value={50} overrides={{ ProgressBar: { styles: { base: { backgroundColor: 'red' } } } }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('ProgressBar.Indicator.base should render correctly', () => {
    const { container } = render(
      <ProgressBar
        value={50}
        overrides={{ ProgressBar: { Indicator: { styles: { base: { backgroundColor: 'red' } } } } }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('ProgressBar.base should render correctly', () => {
    const { container } = render(<ProgressBar value={50} />, {
      theme: { ProgressBar: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('ProgressBar.Indicator.base should render correctly', () => {
    const { container } = render(<ProgressBar value={50} />, {
      theme: { ProgressBar: { Indicator: { styles: { base: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<ProgressBar value={50} />, {
      theme: { ProgressBar: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
