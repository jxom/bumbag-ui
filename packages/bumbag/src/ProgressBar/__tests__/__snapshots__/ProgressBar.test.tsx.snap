// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with ProgressBar props 1`] = `
Object {
  "aria-valuemax": 100,
  "aria-valuemin": 0,
  "aria-valuenow": 50,
  "children": <Box
    className="css-1bsu0i0 css-63it0p css-nkssow fp-Box fp-ProgressBarIndicator"
  />,
  "className": "css-i19h14 css-63it0p css-nkssow fp-Box fp-ProgressBar",
  "role": "progressbar",
}
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 test fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 test fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`overrides ProgressBar.Indicator.base should render correctly 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.emotion-2 {
  background-color: #ff0000;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`overrides ProgressBar.base should render correctly 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-4 {
  background-color: #ff0000;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`props should render correctly with maxValue 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 33.33333333333333%;
}

<div
  aria-valuemax="150"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`props should render correctly with size 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
  height: 2rem;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`theming ProgressBar.Indicator.base should render correctly 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.emotion-2 {
  background-color: #ff0000;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;

exports[`theming ProgressBar.base should render correctly 1`] = `
.emotion-3 {
  color: #574feb !important;
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  height: 100%;
  background-color: #574feb;
  -webkit-transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: width 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  width: 50%;
}

.emotion-4 {
  border-radius: 6px;
  overflow: hidden;
  width: 100%;
  height: 1rem;
  background-color: #eeedfd;
}

.emotion-4 {
  background-color: #ff0000;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="50"
  class="emotion-4 emotion-0 emotion-3 fp-Box fp-ProgressBar"
  role="progressbar"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-ProgressBarIndicator"
  />
</div>
`;
